File: app/src/main/java/com/example/test/mvvmsampleapp/di/FragmentBuildersModule.java
package com.example.test.mvvmsampleapp.di;

import com.example.test.mvvmsampleapp.view.ui.ProjectFragment;
import com.example.test.mvvmsampleapp.view.ui.ProjectListFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class FragmentBuildersModule {
    @ContributesAndroidInjector
    abstract ProjectFragment contributeProjectFragment();

    @ContributesAndroidInjector
    abstract ProjectListFragment contributeProjectListFragment();
}

File: app/src/main/java/com/example/test/mvvmsampleapp/di/FragmentBuildersModule.kt
package com.example.test.mvvmsampleapp.di

import com.example.test.mvvmsampleapp.view.ui.ProjectFragment
import com.example.test.mvvmsampleapp.view.ui.ProjectListFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class FragmentBuildersModule {

    @ContributesAndroidInjector
    internal abstract fun contributeProjectFragment(): ProjectFragment

    @ContributesAndroidInjector
    internal abstract fun contributeProjectListFragment(): ProjectListFragment

}